src/arena.c

```diff
@@ -491,6 +491,20 @@ arena_extent_ralloc_large_expand(tsdn_t *tsdn, arena_t *arena, extent_t *extent,
 	arena_nactive_add(arena, udiff >> LG_PAGE);
 }
 
+static ssize_t
+arena_decay_time_read(arena_t *arena) {
+	arena_decay_time_t dt;
+	dt.u = atomic_read_zu(&arena->decay.time.u);
+	return dt.s;
```

Comment:
Using a union like this is undefined behavior. It may compile correctly today with all common compilers, but there's no reason to believe it will continue to compile correctly.

Is there a reason the atomics can't be used on an ssize_t? Can casting be used instead to produce correct results?